src/main/java/org/itsallcode/holidays/calculator/logic/conditions/DayOfWeekCondition.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * holiday-calculator
+ * Copyright (C) 2021 itsallcode <github@kuhnke.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.itsallcode.holidays.calculator.logic.conditions;
+
+import static java.util.stream.Collectors.joining;
+
+import java.time.DayOfWeek;
+import java.time.MonthDay;
+import java.time.Year;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.itsallcode.holidays.calculator.logic.Formatter;
+
+/**
+ * Example
+ *
+ * <pre>
+ * if DEC 26 is SAT,SUN
+ * </pre>
+ */
+public class DayOfWeekCondition extends Condition {
+
+	public static class UnspecifiedPivotDateException extends RuntimeException {
+		private static final long serialVersionUID = 1L;
+
+		public UnspecifiedPivotDateException(String message) {
+			super(message);
+		}
+	}
+
+	private MonthDay pivot;
+	private final Set<DayOfWeek> daysOfWeek = new HashSet<>();
+
+	public DayOfWeekCondition(DayOfWeek... daysOfWeek) {
+		this(null, daysOfWeek);
+	}
+
+	public DayOfWeekCondition(MonthDay pivot, DayOfWeek... daysOfWeek) {
+		super(false);
+		this.pivot = pivot;
+		this.daysOfWeek.addAll(Arrays.asList(daysOfWeek));
+	}
+
+	@Override
+	public boolean applies(Year year) {
+		if (pivot == null) {
+			throw new UnspecifiedPivotDateException("Cannot apply DayOfWeekCondition with unspecified pivot date.");
```

Comment:
If `pivot` is mandatory, you  should make it a `final` field and check that it is not null in the constructor. If you want to keep `withPivotDate()` for constructing an object, consider using a builder.

Reply:
Next step: try approach as described above.

Reply:
Hi @kaklakariada, I started to use a builder for conditions and it feels quite good, see branch [28-refactor-conditional-holidays](https://github.com/itsallcode/holiday-calculator/tree/Feature/28-refactor-conditional-holidays). 

I plan to use a builder for Holidays, too.
Stay tuned for updates 😄